client/battle/BattleFieldController.cpp

```diff
@@ -338,6 +365,124 @@ std::set<BattleHex> BattleFieldController::getHighlightedHexesForMovementTarget(
 	return {};
 }
 
+std::vector<BattleHex> BattleFieldController::getRangedFullDamageHexes()
+{
+	std::vector<BattleHex> rangedFullDamageHexes; // used for return
+	std::set<BattleHex> battleFieldWithoutSideColumns;
+
+	// if not a hovered arcer unit -> return
+	auto hoveredHex = getHoveredHex();
+	const CStack * hoveredStack = owner.curInt->cb->battleGetStackByPos(hoveredHex, true);
+
+	if (!settings["battle"]["rangedFullDamageLimitHighlightOnHover"].Bool() && !GH.isKeyboardShiftDown())
+		return rangedFullDamageHexes;
+
+	if(!(hoveredStack && hoveredStack->isShooter()))
+		return rangedFullDamageHexes;
+	
+	// construct battlefield grid without the 2 unusable sidecolumns
+	for(auto x = 1; x < GameConstants::BFIELD_WIDTH - 1; ++x)
+	{
+		for(auto y = 0; y < GameConstants::BFIELD_HEIGHT; ++y)
+			battleFieldWithoutSideColumns.insert(BattleHex(x, y));
+	}
+
+	auto rangedFullDamageDistance = GameConstants::BATTLE_PENALTY_DISTANCE;
+
+	// overwrite full ranged damage distance from Additional info field of LIMITED_SHOOTING_RANGE bonus
+	auto bonus = hoveredStack->getBonus(Selector::type()(BonusType::LIMITED_SHOOTING_RANGE));
+	if(bonus != nullptr && bonus->additionalInfo != CAddInfo::NONE)
+		rangedFullDamageDistance = bonus->additionalInfo[0];
+
+	// get only battlefield hexes that are in full range damage distance
+	std::set<BattleHex> fullRangeLimit; 
+	for(auto & hex : battleFieldWithoutSideColumns)
+	{
+		if(BattleHex::getDistance(hoveredHex, hex) <= rangedFullDamageDistance)
+			rangedFullDamageHexes.push_back(hex);
+	}

```

Comment:
You can use `isAvailable` instead of generating `battleFieldWithoutSideColumns` vector
```suggestion
	for(int i = 0; i < GameConstants::BFIELD_SIZE; i++)
	{
		BattleHex hex(i);
		if(hex.isAvailable() && BattleHex::getDistance(hoveredHex, hex) <= rangedFullDamageDistance)
			rangedFullDamageHexes.push_back(hex);
	}

```

Reply:
Good ideea, will change.